lib/streamlit/uploaded_file_manager.py

```diff
@@ -50,6 +50,11 @@ def __init__(self, record: UploadedFileRec):
         self.type = record.type
         self.size = len(record.data)
 
+    def __eq__(self, other):
+        if other is not None:
+            return self.id == other.id
+        return False
+
```

Comment:
To be really correct, `__eq__` should typecheck its argument (and we might as well also type-annotate the function)

```python
def __eq__(self, other: object) -> bool:
  if not isinstance(other, UploadedFile):
    return NotImplemented
  return self.id == other.id
```

(This StackOverflow has more details about the weird "return NotImplemented" line: https://stackoverflow.com/questions/878943/why-return-notimplemented-instead-of-raising-notimplementederror)